### DISCUSSION TITLE
Why are the models retrained for blending and stacking?

### DISCUSSION THREAD

**Discussion by bpisaacoff** [pycaret/pycaret#1258] [Q&A] [answered]
Why are the estimators (estimator in `estimator_list`) retrained for ensembling, blending, or stacking (`ensemble_model()`, `blend_models()`, or `stack_models()`)? 

Those functions accept already created, and therefore trained, models. In order to get metrics on the validation folds should only require making predictions on the validation sets using the already trained models. I'm not sure if this is a bug or if I'm misunderstanding how pycaret is approaching these ensembling methods and would love some help understanding.

Thank you

**Comment by Yard1** [comment, marked as answer]
We are using scikit-learn voting/stacking estimators under the hood. Retraining is the default scikit-learn behavior to ensure robustness.

**Reply by bpisaacoff** [reply]
Got it, thank you. Also I realized I included stacking in my question when I only meant ensembling or blending. It would be very nice to have an option to not retrain for speed purposes.

**Reply by Yard1** [reply]
While that is a good idea, it would require us to modify the sklearn estimators, and we would like to avoid that as much as possible in order to ensure compatibility with other libraries. It may or may not happen in the future.

**Reply by ngupta23** [reply]
I had a similar question when evaluating the time series module using `sktime` and the answer was similar. **[See this](https://github.com/alan-turing-institute/sktime/issues/584#issuecomment-756157813).**

**Reply by bpisaacoff** [reply]
That was an interesting conversation, thanks for sharing
